sponsorship_switzerland/models/contracts.py

```diff
@@ -193,9 +193,7 @@ def contract_active(self):
         Update partner to add the 'Sponsor' category
         """
         super().contract_active()
-        # TODO CO-3284 Replace this check by looking if mail.activity exists
-        need_validation = self.filtered(lambda s: s.partner_id.state != "active")
-        if need_validation:
+        if self.partner_id.activity_ids or self.correspondent_id.activity_ids:
```

Comment:
The method may be multi (self is a recordset) therefore it's forbidden to use `self.partner_id` or `self.correspondent_id`.
You can use the `mapped` method to circumvent this problem.
```suggestion
        if self.mapped("partner_id.activity_ids") or self.mapped("correspondent_id.activity_ids"):
```

Reply:
I (wrongly) assumed these methods would be called on singleton recordset. This is fixed by next commit.